server/app.js

```
@@ -1,5 +1,107 @@
+        return title.includes(searchQuery);
+      });
+    }
+    res.send(json);
+  } catch (error) {
+    res.send(error);
+  }
+});
+
+// Marks ticket as done
+app.post('/api/tickets/:ticketId/done', async (req, res) => {
+  try {
+    const content = await fs.readFile(path);
+    const json = JSON.parse(content);
+    const { ticketId } = req.params;
+    console.log('Query param: ', req.params);
+    console.log('Done', true);
+    let responseJson;
+    json.forEach((ticket, index) => {
+      if (ticket.id === ticketId) {
+        json[index].done = true;
+        responseJson = json[index];
+        responseJson.updated = true;
+      }
+    });
+    await fs.writeFile(path, `${JSON.stringify(json)}`);
+    res.send(responseJson);
+  } catch (error) {
+    res.send(error);
+  }
+});
+
+// Marks ticket as undone
+app.post('/api/tickets/:ticketId/undone', async (req, res) => {
+  try {
+    const content = await fs.readFile(path);
+    const json = JSON.parse(content);
+    const { ticketId } = req.params;
+    console.log('Query param: ', req.params);
+    console.log('Done', false);
+    let responseJson;
+    json.forEach((ticket, index) => {
+      if (ticket.id === ticketId) {
+        json[index].done = false;
+        responseJson = json[index];
+        responseJson.updated = true;
+      }
+    });
+    await fs.writeFile(path, `${JSON.stringify(json)}`);
+    res.send(responseJson);
+  } catch (error) {
+    res.send(error);
+  }
+});
+
+// Add ticket
+app.post('/api/tickets', async (req, res) => {
+  try {
+    const date = new Date();
+    const ticket = req.body;
+    ticket.id = uuid();
+    ticket.creationTime = date.getTime();
+    console.log(ticket);
+    const data = await fs.readFile(path);
+    const dataJson = JSON.parse(data);
+    dataJson.unshift(ticket);
+    await fs.writeFile(path, `${JSON.stringify(dataJson)}`);
+    res.send(dataJson);
+  } catch (error) {
+    res.send(error);
```

This error is not handled by the client